File: Csharp/Lab8/Lab8/Lab8/cng/Model/Collections/neww/Link.cs
namespace Lab8.cng.Model.Collections.neww
{
    public class Link<T>
    {
        private T _element;
        private Link<T> _nextNode, _prevNode;

        public Link(T value, Link<T> prev, Link<T> next) {
            _element = value;
            _nextNode = next;
            _prevNode = prev;
        }

        public T Element
        {
            get => _element;
            set => _element = value;
        }

        public Link<T> NextNode
        {
            get => _nextNode;
            set => _nextNode = value;
        }

        public Link<T> PrevNode
        {
            get => _prevNode;
            set => _prevNode = value;
        }
    }
}

File: Java/Desktop/Saod/Lab3/src/by/bntu/fitr/prostrmk/collections/Link.java
package by.bntu.fitr.prostrmk.collections;

public class Link<T> {


    private T element;
    private Link<T> nextNode, prevNode;

    public Link(T value, Link<T> prev, Link<T> next) {
        element = value;
        nextNode = next;
        prevNode = prev;
    }

    public T getElement() {
        return element;
    }

    public void setElement(T element) {
        this.element = element;
    }

    public Link<T> getNextNode() {
        return nextNode;
    }

    public void setNextNode(Link<T> nextNode) {
        this.nextNode = nextNode;
    }

    public Link<T> getPrevNode() {
        return prevNode;
    }

    public void setPrevNode(Link<T> prevNode) {
        this.prevNode = prevNode;
    }


}
